src/status_im/ethereum/eip681.cljs

```diff
@@ -41,21 +42,31 @@
                {:function-arguments (apply dissoc m valid-native-arguments)}))
       arguments)))
 
-;; TODO add ENS support
-
 (defn parse-uri
-  "Parse a EIP 681 URI as a map (keyword / strings). Parsed map will contain at least the key `address`.
-   Note that values are not decoded and you might need to rely on specific methods for some fields (parse-value, parse-number).
+  "Parse a EIP 681 URI as a map (keyword / strings). Parsed map will contain at least the key `address` 
+   which will be either a valid ENS or Ethereum address.
+   Note that values are not decoded and you might need to rely on specific methods for some fields 
+   (parse-value, parse-number).
    Invalid URI will be parsed as `nil`."
   [s]
   (when (string? s)
     (let [[_ authority-path query] (re-find uri-pattern s)]
       (when authority-path
-        (let [[_ address chain-id function-name] (re-find authority-path-pattern authority-path)]
-          (when (ethereum/address? address)
-            (when-let [arguments (parse-arguments function-name query)]
-              (merge {:address address :chain-id (if chain-id (js/parseInt chain-id) (ethereum/chain-keyword->chain-id :mainnet))}
-                     arguments))))))))
+        (let [[_ raw-address chain-id function-name] (re-find authority-path-pattern authority-path)]
+          (when (or (ethereum/address? raw-address)
+                    (if (string/starts-with? raw-address "pay-")
+                      (let [pay-address (string/replace-first raw-address "pay-" "")]
+                        (or (ens/is-valid-eth-name? pay-address)
+                            (ethereum/address? pay-address)))))
+            (let [address (if (string/starts-with? raw-address "pay-")
+                            (string/replace-first raw-address "pay-" "")
+                            raw-address)]
+              (when-let [arguments (parse-arguments function-name query)]
```

Comment:
Seems like `parse-arguments` are still not validating the parameters, I guess this will be left to whatever calls it?

There is needs for input sanitization in this case? if ,yes, they should be done here or by calling method?

Reply:
This is not a request to change, just opening discussion. @yenda, wdyt?

Reply:
What kind of validation would you expect? is it just `:value` `:gas` and `:gasPrice`?

Reply:
validation of parameters and values, for example, if is uint256, than it should be an valid uint256 format, if is address, it should be an valid address format. 
e.g. 1. `ethereum:pay-a.eth/transfer&uimt256=1e10&address=b.eth` (`uimt256` is invalid); 
2. `ethereum:pay-a.eth/transfer&uint256=1e10&address=Null` (`Null` is invalid); 
3. `ethereum:pay-a.eth/transfer&uint256=1e10&address=myaddress.eth&uint256=1` (the address provided was `myaddress.eth&uint256=1` and it tries to inject additional parameter to parser).

Reply:
I think at least 1-2 should be pretty straight forward. I'm going to update `parse-uri` in the morning and I think it should be pretty straight forward to at least do validation on all the standard required/optional fields. Catching extra parameters might be trickier. Are you only concerned about that with the transfer function?

Reply:
I see that this is out of scope for this PR, we might open an issue about it after this was merged, as this issue and pr is just about the ens support, not validation of fields on EIP681.